### ISSUE TITLE
gwen unable to auto scroll and click on the object

### ISSUE THREAD

**Issue by anshu781126** [gwen-interpreter/gwen-web#46]
And I click ABC

ABC at times is not on screen so we need to scroll and then click; in this scenario gwen is not auto scrolling.

And I scroll to the bottom of ABC
And I click ABC

the above snippet works fine.

Point is gwen is able to recognize ABC object for "And I scroll to the bottom of ABC" DSL in second case but unable to recognize ABC object in first case for "And I click ABC" DSL.

if gwen can recognize ABC object in scroll DSL then why it can't recognize the same in click DSL.

**Comment by bjuric** [comment]
Good question @anshu781126. The scrollTo uses JavaScript to physically move to the element. Click is a webdriver operation that appears to not do the same. So in webdriver land, moving is not an implicit operation (or is at least not consistent across browsers). We could make it implicit but have to consider the side effects. We will look into it.

**Comment by bjuric** [comment]
We've added implicit scrolling for all element actions on the snapshot build below.  Please test this out in your environment and let us know how it goes.
- [gwen-web-2.15.1.2-SNAPSHOT.zip](https://oss.sonatype.org/content/repositories/snapshots/org/gweninterpreter/gwen-web/2.15.1.2-SNAPSHOT/gwen-web-2.15.1.2-SNAPSHOT.zip)

**Comment by anshu781126** [comment]
Its not working for me :(

**Comment by bjuric** [comment]
If you could formulate a JavaScript snippet in your browser console that successful scrolls your element into view, it would be helpful.

**Comment by anshu781126** [comment]
If you meant that I need to write explicit javascript for that then it doesn't add value. Because in this case i will making two calls.. first to javascript and then click. Right now also we have option to do it in steps that too without writing code :).

And I scroll to the bottom of ABC
And I click ABC

**Comment by bjuric** [comment]
It would help me to see what works in your environment so I can then try to build it into anther snapshot for you to try. I don't have access to your environment, so that would help to know what does work.

**Comment by bjuric** [comment]
Hi @anshu781126, I've built another SNAPSHOT that I'd like you to try:
- [gwen-web-2.15.1.3-SNAPSHOT.zip](https://oss.sonatype.org/content/repositories/snapshots/org/gweninterpreter/gwen-web/2.15.1.3-SNAPSHOT/gwen-web-2.15.1.3-SNAPSHOT.zip)

**Comment by bjuric** [comment]
Hi @anshu781126, clicking without scrolling first seems to work on the majority of web sites. We did observe however that an explicit scroll was necessary for some. This is why we provided the scroll DSL. My concern with making it implicit behaviour is that it could impact performance where it is not necessary. I tried making it implicit in the snapshots above, but as you saw it did not work. My only explanation for this is that it does it too fast when inlined with a click. By calling the step explicitly, Gwen gives the browser enough time to respond and synchronise with the web driver.

Sometimes it's hard to find one solution for all. So we went with leaving it up to the user to explicitly do the scrolling where it is necessary.
